File: java/000/LeetCode0053.java
public class LeetCode0053 {
    public int maxSubArray(int[] nums) {
        int max = Integer.MIN_VALUE;
        int cur = 0;
        for (int num : nums) {
            cur += num;
            max = max > cur ? max : cur;
            if (cur <= 0) {
                cur = 0;
            }
        }
        return max;
    }
}

File: cpp/L000/LeetCode0053.cc
#include <vector>
using std::vector;

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int ans{nums[0]};
        int sum{nums[0]};   
        for (std::size_t i{1};i < nums.size();++i) {
            if (sum < 0) {
                sum = nums[i];
            } else {
                sum += nums[i];
            }
            ans = std::max(sum, ans);
        }
        return ans;
    }
};